elastic/kibana
# fixes onFetchContextChanged calls onFetch with outdated values

Closes https://github.com/elastic/kibana/issues/180225

Test instructions
1. start kibana with `yarn start --run-examples`
2. install web logs sample data set
3. create new dashboard, Click "Add panel" and select "Unified search example"
4. Set time range to last 7 days
5. create a filter "bytes >= 15000". Ensure panel updates
6. Click reload, ensure panel re-fetches data
7. Try a bunch of other things in Unified search UI and ensure panel fetches data as expected
